etna/transforms/encoders/categorical.py

```diff
@@ -123,7 +145,16 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         out_column = self._get_column_name()
         result_df = TSDataset.to_flatten(df)
         result_df[out_column] = self.le.transform(result_df[self.in_column], self.strategy)
-        result_df[out_column] = result_df[out_column].astype("category")
+
+        return_type: Union[Type[float], str]
+        if self.return_type == ImputerType.categorical:
+            return_type = "category"
+        elif self.return_type == ImputerType.numeric:
+            return_type = float
+        else:
+            raise ValueError(f"The return_type '{self.return_type}' is not supported")
```

Comment:
Let's put this logic into enum type like it is done in other places.

Reply:
May be rewrite also for `ImputerMode`?

Reply:
Yes, we could do this, but we shouldn't probably change the error type in that case. In a lot of cases the error type was `NotImplementedError`, but for `ImputerMode` we already have `ValueError`.